dev/tasks/linux-packages/apache-arrow/yum/arrow.spec.in

```diff
@@ -135,8 +136,16 @@ cd cpp/build
   -DARROW_WITH_ZSTD=ON \
   -DCMAKE_BUILD_TYPE=$cpp_build_type \
   -DPythonInterp_FIND_VERSION=ON \
-  -DPythonInterp_FIND_VERSION_MAJOR=3
+  -DPythonInterp_FIND_VERSION_MAJOR=3 \
+%if %{use_ninja}
+  -GNinja
+%endif
+
+%if %{use_ninja}
```

Comment:
I discovered that you can use `cmake --build <build_dir> [--target=<target>]` and it'll invoke the builder for you.

Reply:
+1 that's what we do in https://github.com/apache/arrow/blob/master/r/inst/build_arrow_static.sh as well

Reply:
Thanks. I know about it. :-)

But we can't use it here to remove `%if`.

We need to use `cmake` on CentOS 8 and `cmake3` on CentOS 6 and 7. We need `%if` even if we use `cmake --build`.
There is `%cmake3` RPM macro but we can't use it here. Because `%cmake3` expands to `/usr/bin/cmake -D...` on CentOS 8 and `/usr/bin/cmake3 -D...` on CentOS 6 and 7. `-D...` conflicts with `cmake --build`.